Examples/Framework/include/ActsExamples/EventData/Measurement.hpp

```diff
@@ -28,253 +29,357 @@
 
 namespace ActsExamples {
 
-/// A measurement of a variable-size subspace of the full parameters.
-///
-/// @tparam indices_t Parameter index type, determines the full parameter space
-///
-/// The measurement intentionally does not store a pointer/reference to the
-/// reference object in the geometry hierarchy, i.e. the surface or volume. The
-/// reference object can already be identified via the geometry identifier
-/// provided by the source link. Since a measurement **must** be anchored within
-/// the geometry hierarchy, all measurement surfaces and volumes **must**
-/// provide valid geometry identifiers. In all use-cases, e.g. Kalman filtering,
-/// a pointer/reference to the reference object is available before the
-/// measurement is accessed; e.g. the propagator provides the surface pointer
-/// during navigation, which is then used to lookup possible measurements.
-///
-/// The pointed-to geometry object would differ depending on the parameter type.
-/// This means either, that there needs to be an additional variable type or
-/// that a pointer to a base object is stored (requiring a `dynamic_cast` later
-/// on). Both variants add additional complications. Since the geometry object
-/// is not required anyway (as discussed above), not storing it removes all
-/// these complications altogether.
-template <typename indices_t>
-class VariableSizeMeasurement {
+template <typename Derived, std::size_t FullSize, bool ReadOnly>
+class MeasurementProxyBase;
+template <std::size_t FullSize, std::size_t Size, bool ReadOnly>
+class FixedMeasurementProxy;
+template <std::size_t FullSize, bool ReadOnly>
+class VariableMeasurementProxy;
+
+template <std::size_t Size>
+using FixedBoundMeasurementProxy =
+    FixedMeasurementProxy<Acts::eBoundSize, Size, false>;
+template <std::size_t Size>
+using ConstFixedBoundMeasurementProxy =
+    FixedMeasurementProxy<Acts::eBoundSize, Size, true>;
+using VariableBoundMeasurementProxy =
+    VariableMeasurementProxy<Acts::eBoundSize, false>;
+using ConstVariableBoundMeasurementProxy =
+    VariableMeasurementProxy<Acts::eBoundSize, true>;
+
+class MeasurementContainer {
```

Comment:
Even though we are in examples, I think a small class description would be good, since this is a more complex class

Reply:
I added some doc